concepts/infrastructure-reconciliation/README.md

```diff
@@ -1,41 +1,183 @@
+
+### Issuing kubeconfigs
+
+As stated before, the separation of kubeconfigs and limiting access is needed to have a healthy, secure environment. Cluster Managers will provide a possibility to create RBAC-based kubeconfigs in both BYOC and Gardener clusters. This will provide operators with the required access to perform any necessary actions.
+
+The kubeconfigs for specific Kubernetes clusters will be fetched (both from Vault or Gardener) in the runtime and stored only in the components' memory for the limited time needed to issue another (constrained) kubeconfig for a given cluster. Those new kubeconfigs (used by the Lifecycle and Module Managers) will have a short expiration date and will be available in the KCP.
+
+### Subscription Manager and the SubscriptionRequest CR
+
+This is a part that needs to be extracted from the Kyma Environment Broker, there are no significant changes in this area. The main change is introducing a new declarative API in form of the SubscriptionRequest CR:
+```
+apiVersion: control-plane.kyma-project.io/v1alpha1
+kind: SubscriptionRequest
+metadata:
+    name: subscription-request
+    namespace: cluster-manager
+spec:
+    provider: aws
+    subscriptionSelector: "abcd-0001"
+status:
+    gardenerSecretName: subscription-secret
+```
+
+The resource structure is really simple, the Subscription Manager only needs to know the provider and a proper selector for the subscription to be selected. It provides credentials in a form of the Gardener secret name, which can be simply used as a parameter in the Shoot CR.
+
+## Cluster Provisioning Flow
+
+The flow from the user's perspective stays intact, the part that changes is the list of actions needed by KEB to make sure the cluster is up and running.
+
+First of all, after receiving the provisioning request, KEB should create a set of CRs for accordingly to the licence type of the Runtime:
+- For BYOC - ExternalCluster CR
+- For managed - GardenerCluster CR
+
+KEB should deliver all the required parameters for the cluster provisioning, this translates to the minimal infrastructure config for Gardener clusters, and the role/kubeconfig details for both types.
+
+Another thing that KEB would need to create is the Kyma CR for the Runtime, this will either require passing the proper secret reference so that kubeconfig can be accessed by the Lifecycle Manager.
+
+After the CRs are created, the respective Cluster Manager will pick up the provisioning request and will create a new Shoot for the runtime (managed scenario only), and provide a requested kubeconfig.
+
+At this point Lifecycle Manager can step in, it has all required resources to set up the Runtime:
+- running cluster
+- kubeconfig (passed via secret reference)
+- Kyma CR
+
+## Reacting to the kubeconfig rotation
+
+Since Lifecycle Manager and Module Manager need to interact with the provisioned K8S cluster directly, a mechanism for providing them with an up-to-date kubeconfig is needed.
```

Comment:
```suggestion
Since Lifecycle Manager and Module Manager need to interact with the provisioned Kubernetes cluster directly, a mechanism for providing them with an up-to-date kubeconfig is needed.
```